indexer/services/comlink/src/controllers/api/v4/perpetual-positions-controller.ts

```diff
@@ -150,6 +156,143 @@ class PerpetualPositionsController extends Controller {
       }),
     };
   }
+
+  @Get('/parentSubaccountNumber')
+  async listPositionsForParentSubaccount(
+    @Query() address: string,
+      @Query() parentSubaccountNumber: number,
+      @Query() status?: PerpetualPositionStatus[],
+      @Query() limit?: number,
+      @Query() createdBeforeOrAtHeight?: number,
+      @Query() createdBeforeOrAt?: IsoString,
+  ): Promise<PerpetualPositionResponse> {
+    // Get subaccountIds for all child subaccounts of the parent subaccount
+    // Create a record of subaccountId to subaccount number
+    const childIdtoSubaccountNumber: Record<string, number> = {};
+    getChildSubaccountNums(parentSubaccountNumber).forEach(
+      (subaccountNum: number) => {
+        childIdtoSubaccountNumber[SubaccountTable.uuid(address, subaccountNum)] = subaccountNum;
+      },
+    );
+    const childSubaccountIds: string[] = Object.keys(childIdtoSubaccountNumber);
+
+    const [
+      positions,
+      markets,
+    ]: [
+      PerpetualPositionFromDatabase[],
+      MarketFromDatabase[],
+    ] = await Promise.all([
+      PerpetualPositionTable.findAll(
+        {
+          subaccountId: childSubaccountIds,
+          status,
+          limit,
```

Comment:
do we want to remove limit here?

Reply:
This existed in the listPositions API and was brought over as a result. Should this be removed in the original API as well and if so, why?

Reply:
We should keep the limit, this should be the 100 open perpetual positions on the parent-subaccount so it would include positions on child subaccounts.